### PR TITLE
Preserve drawing buffer for gl3d scenes when using Safari 13

### LINKED ISSUE #5158: 3D plots not showing on Safari Version 14.0 when there is transparency in the scene
After updating Safari to version 14.0, you can visit https://plotly.com/javascript/3d-mesh/ and observe the empty 3D scatter plots as follows: 
image

### PR DESCRIPTION
**Pull request by archmoj**
Fixes #5158.

@plotly/plotly_js

### CODE REVIEW

**Comment by archmoj** [comment]
Related info: https://stackoverflow.com/questions/27746091/preservedrawingbuffer-false-is-it-worth-the-effort

File: src/plots/gl3d/scene.js [hunk 1]
```diff
@@ -30,7 +32,48 @@ var createAxesOptions = require('./layout/convert');
...
+        var allParts = ua.split('/');
+        for(var i = 0; i < allParts.length; i++) {
+            var part = allParts[i];
+            if(part.indexOf('Safari') !== -1) {
+                // find Safari version
+                var v = part.split('.')[0];
+                if(isNumeric(v)) v = +v;
+
+                // to fix https://github.com/plotly/plotly.js/issues/5158
+                if(v >= 14) return true;
```
Comment (alexcjohnson) [inline src/plots/gl3d/scene.js:52]: I don't understand `navigator.userAgent`... Here's what I see on my laptop browsers:
```
Chrome:
Mozilla/5.0 (Macintosh; Intel Mac OS X 11_1_0) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/87.0.4280.88 Safari/537.36

Safari:
Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_6) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/14.0.2 Safari/605.1.15

FireFox:
Mozilla/5.0 (Macintosh; Intel Mac OS X 10.16; rv:84.0) Gecko/20100101 Firefox/84.0

Edge:
Mozilla/5.0 (Macintosh; Intel Mac OS X 10_16_0) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/85.0.4183.121 Safari/537.36 Edg/85.0.564.68

Brave:
Mozilla/5.0 (Macintosh; Intel Mac OS X 10_16_0) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/80.0.3987.163 Safari/537.36"
```
So by the test here, everything except FireFox would look like Safari >= 14 - Safari 80, Safari 87... That's probably not what we want.
(Amusingly they all have different ideas what OS version I'm running, and only Chrome is correct. Safari, Apple's own, is the farthest off)

Comment (archmoj) [reply src/plots/gl3d/scene.js]: Good catch. Fixed in the commits below.

**Review by alexcjohnson — APPROVED** [review]
That's quite a list of `userAgent` strings 😳 
Looks great. 💃

**Pull request merged**